infrastructure/src/main/java/org/corfudb/infrastructure/logreplication/replication/receive/LogReplicationSinkManager.java

```diff
@@ -386,4 +402,24 @@ public void shutdown() {
         SNAPSHOT_SYNC,
         LOG_ENTRY_SYNC
     }
+
+    /**
+     * In Snapshot Sync, if the StreamsSnapshotWriter is in the apply phase, the data is not yet
+     * consistent and cannot be read by applications.  Data is always consistent during Log Entry Sync,
+     * so always return true in that case.
+     * @return boolean
+     */
+    private boolean isDataConsistent() {
+        if (RxState.LOG_ENTRY_SYNC == rxState) {
```

Comment:
rxState needs to be thread safe right?
Also, do we really need this distinction? anyways if it is in LOG_ENTRY_SYNC , data_consistent was set to true (maybe we need to confirm that it is set to true even if on re-instantiation it enters directly in log entry sync)